core/src/main/java/io/kestra/core/storages/StorageInterface.java

```diff
@@ -44,15 +49,41 @@ default boolean exists(URI uri) {
         }
     }
 
+    /**
+     * @deprecated
+     * Use {@link #getAttributes(URI)} instead of individual call for every attribute
+     * @param uri
+     * @return
+     * @throws IOException
+     */
+    @Deprecated
     @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
     Long size(URI uri) throws IOException;
 
+    /**
+     * @deprecated
+     * Use {@link #getAttributes(URI)} instead of individual call for every attribute
+     * @param uri
+     * @return
+     * @throws IOException
+     */
     @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
```

Comment:
You miss @Deprecated, moreover I don't know if the @deprecated within javadoc has any sense ?